src/System.Linq/tests/ConcatTests.cs

```diff
@@ -251,20 +251,34 @@ public void CountOfConcatIteratorShouldThrowExceptionOnIntegerOverflow()
             var supposedlyLargeCollection = new DelegateBasedCollection<int> { CountWorker = () => int.MaxValue };
             var tinyCollection = new DelegateBasedCollection<int> { CountWorker = () => 1 };
 
+            Action<Action> assertThrows = (testCode) =>
+            {
+                // The full .NET framework uses unsigned arithmetic summing up collection counts.
+                // See https://github.com/dotnet/corefx/pull/6131.
+                if (PlatformDetection.IsFullFramework)
+                {
+                    testCode();
+                }
+                else
+                {
+                    Assert.Throws<OverflowException>(testCode);
```

Comment:
The longests tests are:

> System.Linq.Tests.TakeTests.LazySkipAllTakenForLargeNumbers(largeNumber: 2147483647): 241.0611664s, 64%

> System.Linq.Tests.MinTests.Min_NullableFloat(source: [NaN, NaN, NaN, NaN, NaN, ...], expected: NaN): 174.0065974s, 47%

> System.Linq.Tests.SelectManyTests.ThrowOverflowExceptionOnConstituentLargeCounts(counts: [2147483647, 1]): 84.0465576s, 22%

> System.Linq.Tests.MinTests.Min_Float(source: [NaN, NaN, NaN, NaN, NaN, ...], expected: NaN): 78.4624312s, 21%

> System.Linq.Tests.MinTests.Min_NullableDouble(source: [NaN, NaN, NaN, NaN, NaN, ...], expected: NaN): 63.7161051s, 17%

> System.Linq.Tests.SelectManyTests.ThrowOverflowExceptionOnConstituentLargeCounts(counts: [2, 2147483646]): 58.0114029s, 16%

> System.Linq.Tests.MinTests.Min_Double(source: [NaN, NaN, NaN, NaN, NaN, ...], expected: NaN): 56.9650254s, 15%

> System.Linq.Tests.RangeTests.LastOrDefault: 54.7322258s, 15%

> System.Linq.Tests.SelectManyTests.ThrowOverflowExceptionOnConstituentLargeCounts(counts: [123, 456, 2147383647, 123456]): 46.4680855s, 12%

> System.Linq.Tests.RangeTests.Last: 21.9727765s, 6%

Percent values are meaningless, as tests can run in parallel